village/src/App.js

```diff
@@ -11,14 +12,59 @@ class App extends Component {
       smurfs: [],
     };
   }
+
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
+
+  addSmurf = (e, smurf) => {
+    e.preventDefault();
+    axios
+      .post('http://localhost:3333/smurfs', smurf)
+      .then(res => {
+        this.setState({
+          smurfs: res.data
+        });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  };
+
+  componentDidMount() {
+    axios
+      .get('http://localhost:3333/smurfs')
+      .then(res => {
+        this.setState({ smurfs: res.data });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
   render() {
     return (
       <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
+
+        <nav>
+          <div className="nav-links">
```

Comment:
Nice job staying semantic with `<nav>`. It can be more semantic by modifying your nav and div to:
```suggestion
      <header>
          <nav className="nav-links">
```